controllers/flinkcluster_updater.go

```diff
@@ -74,10 +74,16 @@ func (updater *_ClusterStatusUpdater) updateClusterStatusIfChanged() error {
 
 func (updater *_ClusterStatusUpdater) deriveClusterStatus() flinkoperatorv1alpha1.FlinkClusterStatus {
 	var status = flinkoperatorv1alpha1.FlinkClusterStatus{}
-	var totalComponents = 3
-	var readyComponents = 0
+	var runningComponents = 0
 	var recordedClusterStatus = &updater.observedState.cluster.Status
 
+	var totalComponents = 0
+	if updater.observedState.cluster.Spec.JobSpec != nil {
+		totalComponents = 4
```

Comment:
Can we make this hardcoded numbers some static variables with some meaningful names?

Reply:
I feel it is better to add comments.